vllm/prompt_adapter/models.py

```diff
@@ -0,0 +1,231 @@
+import logging
+from typing import Any, Callable, Dict, Hashable, List, Optional, Set, Type
+
+from peft.utils import load_peft_weights
+from torch import nn
+
+from vllm.config import PromptAdapterConfig
+from vllm.utils import LRUCache
+
+logger = logging.getLogger(__name__)
+
+_GLOBAL_PROMPT_ADAPTER_ID = 0
+
+
+def get_prompt_adapter_id():
+    global _GLOBAL_PROMPT_ADAPTER_ID
+    _GLOBAL_PROMPT_ADAPTER_ID += 1
+    return _GLOBAL_PROMPT_ADAPTER_ID
+
+
+class PromptAdapterModel(object):
```

Comment:
I think all of this makes a lot of sense, but it makes me wonder if we can promote code reuse between Prompt and Lora managers by eg. making a generic version of those two and then subclass/compose into specialized versions. Right now a lot of code seems duplicated.

Reply:
Yes I agree, we can do that